pkg/deployments/src/actionId.ts

```diff
@@ -0,0 +1,197 @@
+import { FunctionFragment, Interface } from '@ethersproject/abi';
+import { Contract } from '@ethersproject/contracts';
+import fs from 'fs';
+import { padEnd } from 'lodash';
+import path from 'path';
+import logger from './logger';
+import Task, { TaskMode } from './task';
+
+const ACTION_ID_DIRECTORY = path.join(__dirname, '../action-ids');
+
+type ContractActionIdData = { useAdaptor: boolean; exampleContractAddress?: string; actionIds: Record<string, string> };
+type ActionIdInfo = {
+  taskId: string;
+  contractName: string;
+  signature: string;
+  useAdaptor: boolean;
+};
+
+function safeReadJsonFile<T>(filePath: string): Record<string, T> {
+  const fileExists = fs.existsSync(filePath) && fs.statSync(filePath).isFile();
+
+  return fileExists ? JSON.parse(fs.readFileSync(filePath).toString()) : {};
+}
+
+export async function saveActionIds(task: Task, contractName: string, contractAddress?: string): Promise<void> {
+  const { useAdaptor, actionIds } = await getActionIds(task, contractName, contractAddress);
+
+  const actionIdsDir = path.join(ACTION_ID_DIRECTORY, task.network);
+  if (!fs.existsSync(actionIdsDir)) fs.mkdirSync(actionIdsDir, { recursive: true });
+
+  const filePath = path.join(actionIdsDir, 'action-ids.json');
+
+  // Load the existing content if any exists.
+  const newFileContents = safeReadJsonFile<Record<string, ContractActionIdData>>(filePath);
+
+  // Write the new entry.
+  newFileContents[task.id] = newFileContents[task.id] ?? {};
+  newFileContents[task.id][contractName] = { useAdaptor, exampleContractAddress: contractAddress, actionIds };
+
+  fs.writeFileSync(filePath, JSON.stringify(newFileContents, null, 2));
+}
+
+export async function checkActionIds(task: Task): Promise<void> {
+  const filePath = path.join(ACTION_ID_DIRECTORY, task.network, 'action-ids.json');
+  const actionIdFileContents = safeReadJsonFile<Record<string, ContractActionIdData>>(filePath);
+
+  const taskActionIdData = actionIdFileContents[task.id];
+  if (taskActionIdData === undefined) return;
+
+  for (const [contractName, actionIdData] of Object.entries(taskActionIdData)) {
+    const { useAdaptor: expectedUseAdaptor, actionIds: expectedActionIds } = await getActionIds(
+      task,
+      contractName,
+      actionIdData.exampleContractAddress
+    );
```

Comment:
Note that this doesn't check if we're missing action ids in the repo - we need to manually add them as we deploy contracts.